src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,27 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    public function resetExpiringDate()
+    {
+        $this->expiringDate = null;
+    }
+
+    /**
+     * @param \DateTimeImmutable $expiringDate
+     */
+    public function reduceExpiringDate(\DateTimeImmutable $expiringDate)
+    {
+        if (null === $this->expiringDate || $this->expiringDate > $expiringDate) {
```

Comment:
I would do that in a method was `setExpiringDate`. But I didn't want to backport such logic in the class who want to move foward the expiration date.